src/commonMain/kotlin/exchange.dydx.abacus/functional/vault/VaultDepositWithdrawForm.kt

```diff
@@ -38,26 +46,112 @@ data class VaultDepositWithdrawSlippageResponse(
+        textKeyParams: List<ErrorParam>? = null
+    ): ValidationError {
+        return ValidationError(
+            code = code,
+            type = type,
+            fields = fields?.toIList(),
+            action = null,
+            link = null,
+            linkText = null,
+            resources = ErrorResources(
+                title = titleKey?.let { ErrorString(stringKey = it, params = null, localized = null) },
+                text = textKey?.let {
+                    ErrorString(
+                        stringKey = it,
+                        params = textKeyParams?.toIList(),
+                        localized = null,
+                    )
+                },
+                action = null,
+            ),
+        )
+    }
 
-@JsExport
-@Serializable
-enum class VaultFormValidationErrorType {
-    AMOUNT_EMPTY,
-    DEPOSIT_TOO_HIGH,
-    WITHDRAW_TOO_HIGH,
-    WITHDRAWING_LOCKED_BALANCE,
-    SLIPPAGE_TOO_HIGH,
-    MUST_ACK_SLIPPAGE,
-    VAULT_ACCOUNT_MISSING,
-    ACCOUNT_DATA_MISSING,
-    SLIPPAGE_RESPONSE_MISSING,
-    SLIPPAGE_RESPONSE_WRONG_SHARES,
+    fun amountEmpty(operation: VaultFormAction) = createError(
+        code = "AMOUNT_EMPTY",
+        type = ErrorType.error,
+        fields = listOf("amount"),
+        titleKey = if (operation == VaultFormAction.DEPOSIT) {
+            "APP.VAULTS.ENTER_AMOUNT_TO_DEPOSIT"
+        } else {
+            "APP.VAULTS.ENTER_AMOUNT_TO_WITHDRAW"
+        },
+    )
+
+    fun accountDataMissing(canViewAccount: Boolean?) = createError(
+        code = "ACCOUNT_DATA_MISSING",
+        type = ErrorType.error,
+        titleKey = if (canViewAccount != null && canViewAccount) {
+            "APP.GENERAL.NOT_ALLOWED"
+        } else {
+            "APP.GENERAL.CONNECT_WALLET"
+        },
+    )
+
+    fun depositTooHigh() = createError(
+        code = "DEPOSIT_TOO_HIGH",
+        type = ErrorType.error,
+        fields = listOf("amount"),
+        titleKey = "APP.TRADE.MODIFY_SIZE_FIELD",
+        textKey = "APP.VAULTS.DEPOSIT_TOO_HIGH",
+    )
+
+    fun withdrawTooHigh() = createError(
+        code = "WITHDRAW_TOO_HIGH",
+        type = ErrorType.error,
+        fields = listOf("amount"),
+        titleKey = "APP.TRADE.MODIFY_SIZE_FIELD",
+        textKey = "APP.VAULTS.WITHDRAW_TOO_HIGH",
+    )
+
+    fun withdrawingLockedBalance() = createError(
+        code = "WITHDRAWING_LOCKED_BALANCE",
+        type = ErrorType.error,
+        fields = listOf("amount"),
+        titleKey = "APP.TRADE.MODIFY_SIZE_FIELD",
+        textKey = "APP.VAULTS.WITHDRAW_TOO_HIGH",
+    )
+
+    fun slippageTooHigh(slippagePercent: Double) = createError(
+        code = "SLIPPAGE_TOO_HIGH",
+        type = ErrorType.warning,
+        textKey = "APP.VAULTS.SLIPPAGE_WARNING",
+        textKeyParams = listOf(
+            ErrorParam(key = "AMOUNT", value = slippagePercent.format(4), format = ErrorFormat.Percent),
+            ErrorParam(key = "LINK", value = "", format = null),
```

Comment:
I actually don't know where to store the url or how to pass the anchor text and stuff. Can we just trust the mobile clients will handle this?

Reply:
Sure.. we can update it when working on the mobile client!